### PR TITLE
optimize: String comparison uses equalsIgnoreCase()

### PR DESCRIPTION
**Pull request by github-ganyu**
### Ⅰ. Describe what this PR did
I optimized some of the code: String comparison uses equalsIgnoreCase()

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/xa/ExecuteTemplateXA.java [hunk 1]
```diff
@@ -76,7 +76,7 @@
                         "Failed to commit xa branch of " + connectionProxyXA.xid + ") since " + ex.getMessage(),
                         ex);
                     // XA End & Rollback
-                    if (!(ex instanceof SQLException) || AbstractConnectionProxyXA.SQLSTATE_XA_NOT_END != ((SQLException) ex).getSQLState()) {
+                    if (!(ex instanceof SQLException) || !AbstractConnectionProxyXA.SQLSTATE_XA_NOT_END.equals(((SQLException) ex).getSQLState())) {
```
Comment (a364176773) [inline rm-datasource/src/main/java/io/seata/rm/datasource/xa/ExecuteTemplateXA.java:79]: equalsIgnoreCase

**Comment by a364176773** [comment]
标题以optimize: 开头,并把pr改动登记到changes文件夹中的2个md

**Review by slievrly — APPROVED** [review]
LGTM

**Review by l81893521 — APPROVED** [review]
LGTM

**Pull request merged**